components/patient-data/migrations/src/main/java/org/phenotips/data/internal/R70190PhenoTips1280DataMigration.java

```diff
@@ -0,0 +1,290 @@
+
+    private static final String STATUS_NAME = "status";
+
+    private static final String OR = "' or o.className = '";
+
+    private static final EntityReference PATIENT_CLASS = new EntityReference("PatientClass", EntityType.DOCUMENT,
+        Constants.CODE_SPACE_REFERENCE);
+
+    private static final EntityReference INVESTIGATION_CLASS = new EntityReference("InvestigationClass",
+        EntityType.DOCUMENT,
+        Constants.CODE_SPACE_REFERENCE);
+
+    private static final EntityReference GENE_CLASS = new EntityReference("GeneClass", EntityType.DOCUMENT,
+        Constants.CODE_SPACE_REFERENCE);
+
+    private static final EntityReference REJECTED_CLASS = new EntityReference("RejectedGenesClass",
+        EntityType.DOCUMENT,
+        Constants.CODE_SPACE_REFERENCE);
+
+    /** Resolves unprefixed document names to the current wiki. */
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<String> resolver;
+
+    /** Serializes the class name without the wiki prefix, to be used in the database query. */
+    @Inject
+    @Named("compactwiki")
+    private EntityReferenceSerializer<String> serializer;
+
+    /** Resolves class names to the current wiki. */
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<EntityReference> entityResolver;
+
+    @Override
+    public String getDescription()
+    {
+        return "Migrate all existing gene values to the GeneClass objects";
+    }
+
+    @Override
+    public XWikiDBVersion getVersion()
+    {
+        return new XWikiDBVersion(70190);
+    }
+
+    @Override
+    public void hibernateMigrate() throws DataMigrationException, XWikiException
+    {
+        getStore().executeWrite(getXWikiContext(), this);
+    }
+
+    @Override
+    public Object doInHibernate(Session session) throws HibernateException, XWikiException
+    {
+        XWikiContext context = getXWikiContext();
+        XWiki xwiki = context.getWiki();
+        DocumentReference patientClassReference =
+            R70190PhenoTips1280DataMigration.this.entityResolver.resolve(PATIENT_CLASS);
+        DocumentReference investigationClassReference =
+            R70190PhenoTips1280DataMigration.this.entityResolver.resolve(INVESTIGATION_CLASS);
+        DocumentReference geneClassReference =
+            R70190PhenoTips1280DataMigration.this.entityResolver.resolve(GENE_CLASS);
+        DocumentReference rejectedGenesClassReference =
+            R70190PhenoTips1280DataMigration.this.entityResolver.resolve(REJECTED_CLASS);
+
+        Query q =
+            session.createQuery("select distinct o.name from BaseObject o, StringProperty p where o.className = '"
+                + this.serializer.serialize(geneClassReference) + OR
+                + this.serializer.serialize(rejectedGenesClassReference) + OR
+                + this.serializer.serialize(patientClassReference) + "' and p.id.id = o.id and p.id.name = '"
+                + SOLVED_NAME + "' and p.value <> ''");
```

Comment:
In other words, we are collecting all records from INVESTIGATION_CLASS (candidate genes) and REJECTED_CLASS (rejected genes), but from PATIENT_CLASS we need only one property - SOLVED_NAME (solved genes)